spec/factories.rb

```diff
@@ -540,6 +540,56 @@
       end
     end
 
+    trait :with_structured_rejection_reasons do
+      with_rejection_by_default
+      structured_rejection_reasons do
+        {
+          course_full_y_n: 'No',
+          candidate_behaviour_y_n: 'Yes',
+          candidate_behaviour_other: 'Persistent scratching',
+          candidate_behaviour_what_to_improve: 'Not scratch so much',
+          candidate_behaviour_what_did_the_candidate_do: %w[didnt_reply_to_interview_offer didnt_attend_interview other],
+          honesty_and_professionalism_y_n: 'Yes',
+          honesty_and_professionalism_concerns_other_details: nil,
+          honesty_and_professionalism_concerns: %w[information_false_or_inaccurate references],
+          honesty_and_professionalism_concerns_plagiarism_details: nil,
+          honesty_and_professionalism_concerns_references_details: 'Clearly not a popular student',
+          honesty_and_professionalism_concerns_information_false_or_inaccurate_details: 'Fake news',
+          offered_on_another_course_y_n: 'Yes',
+          offered_on_another_course_details: nil,
+          performance_at_interview_y_n: 'Yes',
+          performance_at_interview_what_to_improve: 'Be fully dressed',
+          qualifications_y_n: 'Yes',
+          qualifications_other_details: 'All the other stuff',
+          qualifications_which_qualifications: %w[no_english_gcse other],
+          quality_of_application_y_n: 'Yes',
+          quality_of_application_other_details: 'Lights on but nobody home',
+          quality_of_application_other_what_to_improve: 'Study harder',
+          quality_of_application_which_parts_needed_improvement: %w[personal_statement subject_knowledge other],
+          quality_of_application_subject_knowledge_what_to_improve: 'Claiming to be the \'world\'s leading expert\' seemed a bit strong',
+          quality_of_application_personal_statement_what_to_improve: 'Use a spellchecker',
+          safeguarding_y_n: 'Yes',
+          safeguarding_concerns: %w[other],
+          safeguarding_concerns_other_details: nil,
```

Comment:
I think we should make the factory consistent with the business rules. So safeguarding should be `No` if we're not going to assign one of the corresponding values. At the moment the UI is in an invalid state with a factory created object.

![image](https://user-images.githubusercontent.com/42515961/102218648-318a4200-3ed6-11eb-96ce-5af135b1dcb8.png)